File: ContainsDuplicate.java
import java.util.ArrayList;

/**
 * Problem: #217 Contains Duplicate
 *
 * Idea:
 *
 * Solutions:
 *  singleNumberHash: using a hash map to store unique characters found.
 *  singleNumber:
 *
 * Note: #canbefaster
 */
public class ContainsDuplicate {
    public boolean containsDuplicate(int[] nums) {
        ArrayList<Integer> list = new ArrayList<>();
        for (int num : nums) {
            if (list.contains(num)) return true;
            list.add(num);
        }
        return false;
    }
}

File: contains_duplicate.rb
# Problem: # 217 Contains Duplicate
# Idea: find wether there is a difference in size between the uniq arry and original array
# Solutions: use .uniq.size()

# @param {Integer[]} nums
# @return {Boolean}
def contains_duplicate(nums)
    return nums.uniq.size() != nums.size()
end
